src/MessagePack.UnityClient/Assets/Scripts/MessagePack/MessagePackCompression.cs

```diff
@@ -25,5 +25,33 @@ public enum MessagePackCompression
         /// See also ThisLibraryExtensionTypeCodes.LZ4
         /// </devremarks>
         Lz4Block,
+
+        /// <summary>
+        /// Compresses an entire msgpack sequence as a multi contiguous block.
+        /// This is the fastest and results in the best LZ4 compression,
+        /// at the cost of copying the entire sequence when necessary to get contiguous memory.
+        /// </summary>
+        /// <remarks>
+        /// Uses msgpack type code ext98 in array.
+        /// </remarks>
+        /// <devremarks>
+        /// See also ThisLibraryExtensionTypeCodes.LZ4Contiguous
+        /// </devremarks>
+        Lz4ContiguousBlock,
+    }
+
+#pragma warning disable SA1649 // File name should match first type name
+
+    /// <summary>
+    /// Extensions for <see cref="MessagePackCompression"/>.
+    /// </summary>
+    public static class MessagePackCompressionExtensions
```

Comment:
Can we keep this internal? So far the value is quite small IMO, and in general we should consider the cost of each new public API to be significant since it can limit options in the future and increase the risk of breaking changes being necessary later on. So I try to avoid new public API unless there's compelling value. In this case, it's a simple enum value check that anyone could figure out, so  I would remove or make internal.

Reply:
Certainly it looks good with Internal.